docs/guides/git/index.md

```diff
@@ -0,0 +1,473 @@
+When you first register on Moodle Tracker you can not assign issues to yourself or send them for peer review. You will be added to the developers group after your first bug fix is integrated. Before that, just comment on the issue with a link to your branch and a component lead, or another developer, will send the issue for peer review for you.
+:::
+
+## Installing Git on your computer
+
+### Installing Git
+
+Most Linux distributions have Git available as a package to install.
+
+#### Debian/Ubuntu
+
+```
+sudo apt-get install git
+```
+
+#### MacOS (Homebrew)
+
+```
+brew install git
+```
+
+#### Windows
+
+Download the latest Git installer at [Git for Windows](https://gitforwindows.org/).
+
+## Configuring Git
+
+After installation, set your name and email. Your name and email will become part of your commits and can't be changed later once your commits are accepted into Moodle code. Therefore we ask contributors to use their real names written in with proper capitalisation. E.g. "John Smith" and not "john smith", or even "john5677".
+
+```
+    git config --global user.name "Your Name"
+    git config --global user.email yourmail@domain.tld
+```
+
+Unless you are the repository maintainer, it is wise to set your Git to not push changes in file permissions:
+
+```
+    git config --global core.filemode false
+```
+
+It's recommended to verify that the your Git installation is not performing any transformation between LFs and CRLFs. Moodle use **only LFs** and you should configure your editor/IDE the same way. Note that having any "magic" enabled is known to cause [problems with unit tests](https://docs.moodle.org/dev/Common_unit_test_problems#The_test_file_.22evolution.test.22_should_not_contain_section_named_.22.5Blots_of_content.5D.22) execution. We recommend you to set:
+
+```
+    git config --global core.autocrlf false
+```
+
+## Setting-up your public repository
+
+1. Go to [Github](http://github.com/) and create an account.
+
+2. Go to the official [Moodle Github repository](http://github.com/moodle/moodle) and click on the Fork button. You now have your own Github Moodle repository.
+
+3. Set up your SSH public key so you can push to your Github Moodle repository from your local Moodle repository. Over at Github, navigate to your Settings page and look for a section called **SSH and GPG keys**. Configuring your SSH key can be a little tricky, but following [Github's SSH setup steps](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent#adding-your-ssh-key-to-the-ssh-agent) will ensure the hardest part of this setup is done correctly.
+
+4. Optionally, you can set up [Github Actions](/general/development/tools/gha) to automatically test your code. This provides the advantage of catching issues without having to set up a testing environment locally and provides helpful feedback before you submit your code for review.
+
+## Setting-up your local repository
```

Comment:
It may be worth linking the MDK documentation here as an alternative.

For those contributing to core, mdk is probably easier in the long-run.

Reply:
I have decided to put a mention of MDK at the top of the page and periodically reference the benefits of using MDK as an alternative throughout.